### PR TITLE
Add auto-generated product layout

### LINKED ISSUE #135: Add auto-rendered product layout
Add auto-rendered product layout where the product data is rendered based on [product drop](https://www.voog.com/developers/markup/ecommerce-objects/product) data.

It should look like current "Product" layout. Use new [image](https://www.voog.com/developers/markup/tags/image) or [image_data](https://www.voog.com/developers/markup/tags/image-data) tag for product cover image.

It content type must be `product`:

```json
{
  "content_type": "product",
  "component": false,
  "file": "layouts/auto_generated_product.tpl",
  "layout_name": "auto_generated_product",
  "title": "Auto-generated product"
}
```

See also buy button rendering example in [Nuuk design](https://github.com/Voog/design-nuuk/blob/master/components/buy-button.tpl).

## TODO

* [ ] Add "Auto-generated product" layout
* [ ] Check and improve "[product list](https://www.voog.com/support/online-store/product-list-content-area)" widget styles when needed.

### PR DESCRIPTION
**Pull request by pxska**
* Add auto-generated product layout.
* Add a browser hack for Chrome to render crispier images.
* Also fix underline in product widget grid.

Closes #135.

### CODE REVIEW

File: layouts/auto_generated_product.tpl [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+              <div class="content-gallery content-area js-product-gallery" data-search-indexing-allowed="true">
+                {% content bind=product name="gallery" %}
+              </div>
+            {%- endif -%}
+
+          </div>
+        </div>
+
+        <div class="flex-col">
+          <div class="content-body product-content-right">
```
Comment (PeeterPaal) [inline layouts/auto_generated_product.tpl:66]: `js-product-content-right` class should be added here

File: layouts/auto_generated_product.tpl [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            <div class="content-item-title content-formatted" data-search-indexing-allowed="true">
+              <h1>{%- editable product.name -%}</h1>
+            </div>
+
+            <div class="product-price">
+              {%- if product.price_max_with_tax != product.price_min_with_tax -%}
+                {{ product.price_min_with_tax | money_with_currency: product.currency -}}
+                <span class="product-price-divider">–</span>
```
Comment (PeeterPaal) [inline layouts/auto_generated_product.tpl:74]: Perhaps add some left and right margin for the price divider

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Looks pretty good, but a few comments too

- [x] Currently gallery is not moved under the buy button in mobile view

File: layouts/auto_generated_product.tpl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              <div class="content-gallery content-area js-product-gallery" data-search-indexing-allowed="true">
+                {% content bind=product name="gallery" %}
+              </div>
+            {%- endif -%}
+
+          </div>
+        </div>
+
+        <div class="flex-col">
+          <div class="content-body js-product-content-right">
```
Comment (PeeterPaal) [inline layouts/auto_generated_product.tpl:66]: The product right content is not sticky on scroll. The style rule checks for `product-content-right` class, so let's re-add that
```suggestion
          <div class="content-body js-product-content-right product-content-right">
```

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Just one little thing

**Review by PeeterPaal — APPROVED** [review]
Great job!

**Review by tanelj — APPROVED** [review]
Nice work!

**Review by tanelj — CHANGES_REQUESTED** [review]
Nice, but there is one typo.

**Review by PeeterPaal — APPROVED** [review]
Looks good!

**Pull request merged**